core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1073,201 +1073,210 @@ class KafkaApis(val requestChannel: RequestChannel,
    * Handle a shareFetch request
    */
   def handleShareFetchRequest(request: RequestChannel.Request): Unit = {
-    val versionId = request.header.apiVersion
-    val clientId = request.header.clientId
     val shareFetchRequest = request.body[ShareFetchRequest]
-    val groupId = shareFetchRequest.data().groupId()
-    val topicNames = metadataCache.topicIdsToNames()
-    val shareFetchData = shareFetchRequest.shareFetchData(topicNames)
-    val forgottenTopics = shareFetchRequest.forgottenTopics(topicNames)
-    val sharePartitionManager : SharePartitionManager = sharePartitionManagerOption match {
-      case Some(manager) => manager
-      case None => throw new IllegalStateException("ShareFetchRequest received but SharePartitionManager is not initialized")
-    }
-    // TODO : replace this initialization when share fetch session metadata is sent by the client in the shareFetchRequest
-    val newReqMetadata : ShareFetchMetadata = new ShareFetchMetadata(Uuid.ZERO_UUID, -1)
-    val shareFetchContext = sharePartitionManager.newContext(shareFetchData, forgottenTopics, topicNames, newReqMetadata)
-    val erroneous = mutable.Map[TopicIdPartition, ShareFetchResponseData.PartitionData]()
-    val interesting = mutable.ArrayBuffer[TopicIdPartition]()
-    // Regular Kafka consumers need READ permission on each partition they are fetching.
-    val partitionDatas = new mutable.ArrayBuffer[(TopicIdPartition, ShareFetchRequest.SharePartitionData)]
-    val cachedPartitionData = shareFetchContext.cachedPartitions().asScala
-    cachedPartitionData.foreach{ case (topicIdPartition: TopicIdPartition, sharePartitionData: ShareFetchRequest.SharePartitionData) =>
-      if (topicIdPartition.topic == null)
-        erroneous += topicIdPartition -> ShareFetchResponse.partitionResponse(topicIdPartition, Errors.UNKNOWN_TOPIC_ID)
-      else
-        partitionDatas += topicIdPartition -> sharePartitionData
-    }
-    val authorizedTopics = authHelper.filterByAuthorized(request.context, READ, TOPIC, partitionDatas)(_._1.topicPartition.topic)
 
-    partitionDatas.foreach { case (topicIdPartition, data) =>
-      if (!authorizedTopics.contains(topicIdPartition.topic))
-        erroneous += topicIdPartition -> ShareFetchResponse.partitionResponse(topicIdPartition, Errors.TOPIC_AUTHORIZATION_FAILED)
-      else if (!metadataCache.contains(topicIdPartition.topicPartition))
-        erroneous += topicIdPartition -> ShareFetchResponse.partitionResponse(topicIdPartition, Errors.UNKNOWN_TOPIC_OR_PARTITION)
-      else
-        interesting += topicIdPartition
+    if (!config.isShareGroupEnabled) {
+      // The API is not supported when the configuration `group.share.enable` has not been set
+      requestHelper.sendMaybeThrottle(request, shareFetchRequest.getErrorResponse(Errors.UNSUPPORTED_VERSION.exception))
+      CompletableFuture.completedFuture[Unit](())
```

Comment:
rather that so much if/else why not to call `return` from above if itself.

Reply:
agreed, I was not using `return` earlier after reading this documentation https://www.baeldung.com/scala/return-keyword#referential-transparency but realized we can't apply this thought process while returning void. Also, I see that it is being used at multiple places in KafkaApis.scala, so it should be fine. Thanks for the suggestion.